tests/storage_test.go

```diff
@@ -171,6 +174,48 @@ var _ = Describe("Storage", func() {
 			}, 240)
 		})
 
+		Context("With an emptyDisk defined", func() {
+			// The following case is mostly similar to the alpine PVC test above, except using different VM.
+			It("should create an emptyDisk with the right capacity", func(done Done) {
+
+				// Start the VM with the empty disk attached
+				vm := tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "echo hi!")
+				vm.Spec.Domain.Devices.Disks = append(vm.Spec.Domain.Devices.Disks, v1.Disk{
+					Name:       "emptydisk1",
+					VolumeName: "emptydiskvolume1",
+					DiskDevice: v1.DiskDevice{
+						Disk: &v1.DiskTarget{
+							Bus: "virtio",
+						},
+					},
+				})
+				vm.Spec.Volumes = append(vm.Spec.Volumes, v1.Volume{
+					Name: "emptydiskvolume1",
+					VolumeSource: v1.VolumeSource{
+						EmptyDisk: &v1.EmptyDiskSource{
+							Capacity: resource.MustParse("2Gi"),
+						},
+					},
+				})
+				RunVMAndExpectLaunch(vm, false, 45)
+
+				expecter, err := tests.LoggedInCirrosExpecter(vm)
+				defer expecter.Close()
+				Expect(err).To(BeNil())
+
+				By("Checking that /dev/vdc has a capacity of 2Gi")
+				res, err := expecter.ExpectBatch([]expect.Batcher{
+					&expect.BSnd{S: "sudo blockdev --getsize64 /dev/vdc\n"},
+					&expect.BExp{R: "2147483648"}, // 2Gi in bytes
```

Comment:
I think it be worth creating a filesystem and verifying that the volume is writable too.

Reply:
Done. The test now creates an ext4 file system on the device.